### ISSUE TITLE
Symlinks break sync function

### ISSUE THREAD

**Issue by grasponcrypto** [photoview/photoview#465]
**Describe the bug**
I have a few symlinks within my directories, this causes the sync function to fail.  

**To Reproduce**
1. In your /photos directory create a symlink `ln -s /photos/dirName /photos/linkName`
2. Click on 'scan'
3. See error

**Expected behavior**
Skip symlinks and/or continue past the error

**Screenshots**

**Your environment:**
Feel free to remove and add what you think might be useful.

 - Are you using Docker? [yes]
 - Database [mariadb 10.5]
 - Version [undefined]
 - Build Date [2021-07-22T16:25:13Z]

**Comment by viktorstrate** [comment]
The best solution might be to make follow symlinks an opt-in, as detecting cycles might be complicated.

**Comment by kkovaletp** [comment]
@nakal do you know that you could achieve the same result, as with symlinks, by using multiple volumes for the Photoview media location (1st volume to map the root folder of the main part of your media library, and then each new volume for the root folder of each originally symlinked external media lib to be mounted in some folder inside the main lib inside the docker)?

While this is a pure workaround and it does nothing with the mentioned bug, it might help you in practice

**Comment by nakal** [comment]
@kkovaletp Yes, thank you for the useful hint. I'd maybe be able to find this workaround, too, if PhotoView just reported an error/warning accessing the softlink.

**Comment by niklas88** [comment]
I too just lost a few hours on this mostly because there is still no useful error shown on the command line. 
It's a shame because otherwise this is an amazing app and it works great when it does scan photos.

It's especially sad because the README proudly states:

**When your media has been scanned they show up on the website, organised in the same way as on the filesystem.**

Well on my filesystem the phtos are oganized as:
```
2023
  01
    photo_in_january_2023.jpg
    …
  02
    photo_in_feburary_2023.heic
    …
  …
2022
  ...
New Year's Party
   photo_in__january_2023 -> ../2023/01/photo_in_january_2023.jpg
```
Which is quite practical in that it gives you both timeline and album access even
in normal file browsers while not needing any copies. It's also pretty much
the format  the [Google Photos Takeout Helper](https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper)
would generate once they implement my request to use relative symlinks which
are still valid after mounting in docker vs absolute symlinks which break.
No sadly this then doesn't work at all if  you break on symlinks.

Most weirdly though just scanning the albums (e.g. New Year's Party above) did work
for me!?!?

**Comment by kkovaletp** [comment]
@niklas88 I'm just curious: what will be in case you replace your softlinks with hardlinks - will it still work for you? will it start working after mounting into the Photoview container? Do you see some negative impact for this solution?

**Comment by niklas88** [comment]
Since I'm using btrfs I could use `cp --reflink` but this is much less portable than simple relative symlinks also with symlinks you can easily see where the "real" files are supposed to be. Also with photoview the symlinks worked when album folder was added explicitly (with the whole folder being volume mounted into the docker container) but not when scanning the whole folder.

**Comment by mdaeron** [comment]
> Well on my filesystem my photos are oganized as...

I tried exactly the same scheme on my system (Unraid 6.12 with photos stored in a zfs pool; originals archived by date and albums containing soft links).

Weirdly enough it seemed to work initially, but I've removed and added some originals and some symlinks, and the album doesn't show up anymore. No visible error message in logs.

Is there an official way to make this work? Hard links?

**Comment by kkovaletp** [comment]
@mdaeron, well, there is no official workaround, except for avoiding symlink usage.
If all your media collection (the originals and the albums) are located on the same physical partition, the best solution would be to use hardlinks instead.

**Comment by mdaeron** [comment]
> @mdaeron, well, there is no official workaround, except for avoiding symlink usage. If all your media collection (the originals and the albums) are located on the same physical partition, the best solution would be to use hardlinks instead.

Thanks, I understand. For now I'm testing a different approach, with one `archive` user and another `albums` user. This seems to avoid duplicate issues, and the use cases for the two different users are clearly separated.
